python/cudf/cudf/tests/test_parquet.py

```diff
@@ -3180,3 +3180,112 @@ def test_parquet_reader_zstd_huff_tables(datadir):
     expected = pa.parquet.read_table(fname).to_pandas()
     actual = cudf.read_parquet(fname)
     assert_eq(actual, expected)
+
+
+def test_parquet_reader_duration_types():
+    # arrow schema is used to faithfully round trip duration
+    # types (timedelta64) across Parquet read and write.
+    pdf = pd.DataFrame(
+        {
+            "s": pd.Series([1234, 3456, 32442], dtype="timedelta64[s]"),
+            "ms": pd.Series([1234, 3456, 32442], dtype="timedelta64[ms]"),
+            "us": pd.Series([1234, 3456, 32442], dtype="timedelta64[us]"),
+            "ns": pd.Series([1234, 3456, 32442], dtype="timedelta64[ns]"),
+        }
+    )
+
+    # Convert all pdf dtypes to int64
+    pdf2 = pdf.astype("int64")
+
+    # Write parquet with arrow for now (to write arrow:schema)
+    buffer = BytesIO()
+    pdf.to_parquet(buffer, engine="pyarrow")
+
+    # Read parquet with arrow schema
+    got = cudf.read_parquet(buffer)
+    # Convert to cudf table for an apple to apple comparison
+    expected = cudf.from_pandas(pdf)
+
+    # Check results for reader with schema
+    assert_eq(expected.dtypes, got.dtypes)
+    assert_eq(expected, got)
+
+    # Read parquet without arrow schema
+    got2 = cudf.read_parquet(buffer, use_arrow_schema=False)
+    # Convert to cudf table for an apple to apple comparison
+    expected2 = cudf.from_pandas(pdf2)
+
+    # Check results for reader without schema
+    assert_eq(expected2.dtypes, got2.dtypes)
+    assert_eq(expected2, got2)
+
+
+def test_parquet_reader_roundtrip_with_arrow_schema():
+    # Ensure that the nested types are faithfully being roundtripped
+    # across Parquet with arrow schema present
+
+    # data frame with several types that round trip perfectly
+    # without arrow schema as well
+    expected = cudf.DataFrame(
+        {
+            "int64": cudf.Series([1234, 123, 4123], dtype="int64"),
+            "list": list([[1, 2], [1, 2], [1, 2]]),
+            "datetime": cudf.Series([1234, 123, 4123], dtype="datetime64[ms]"),
+            "map": cudf.Series(["cat", "dog", "cow"]).map(
+                {"cat": "kitten", "dog": "puppy", "lion": "cub"}
```

Comment:
I've got an amazing question here: should this cow be a lion? :D

Reply:
🤣🤣🤣 Saw this just now. I fixed it in the updated tests. Caught while updating them